Group2 filesystem mock

agardella opened the pull request:
@captain695 @cburk

Commit:
Added dummyfilesystem folder with class file with hashmap

Commit:
Added method signatures for dummy filesystem

Commit:
Fixed method declaration formatting

Commit:
Wrote the getBlock function and changed hashmap of blocks from ints to longs
i Please enter the commit message for your changes. Lines starting

Commit:
Dummy FS code (tests are broken!)

Commit:
Fixed syntax errors in allocate block method

Commit:
RM block stuff compiling

Commit:
Add error checking

Commit:
Merge branch 'group2-filesystem-mock' of https://github.com/Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
Allocate blocks

Commit:
Fixed string errors in dummy-filesystem.cc

Commit:
Merge branch 'group2-filesystem-mock' of github.com:Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
reformatted testing

Commit:
Merge branch 'group2-filesystem-mock' of github.com:Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
Added a test for can add block

Commit:
Allocating blocks

Commit:
Changed print statements in tests.cc to use cout

Commit:
Changed print statements to use cout

Commit:
Merge branch 'group2-filesystem-mock' of github.com:Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
fixed remove block test

Commit:
Fixed test to read in correct block length

Commit:
Merge branch 'group2-filesystem-mock' of github.com:Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
added test for removing nonexistant block

Commit:
Clean up files

Commit:
Merge branch 'develop' into group2-filesystem-mock

Commit:
Merge branch 'group2-filesystem-mock' of https://github.com/Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
Add CMakeLists for fs

Commit:
Merging my test adding (unfinished) with allisons commit

Commit:
Merge branch 'group2-filesystem-mock' of github.com:Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
Added test for checking if correct error response when one tries to add a file that already exists

Commit:
Merge branch 'group2-filesystem-mock' of github.com:Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
Delete tests.cc

jfking commented:
What was your reasoning for using the DummyFS name instead of just naming it NativeFS or something? If it's named the latter (or something like it) we can change the implementation code without changing the code that uses it.

agardella commented:
Good point, we'll change that.

efrederic commented on dummy-filesystem/dummy-filesystem.cc:
```diff
@@ -0,0 +1,96 @@
+
+	//Copy to a char*, which erase and remove need
+	char *fileNameFmtd = const_cast<char*>(fileName.c_str());
+	blockMap.erase(iter);
```
Looks like this will remove the block from the map even if there was an error deleting the file

kyler-m commented:
Not that it matters much, but stdout and stderr seem to be used interchangeably between the functions for error messages. Might want to change them all to stderr? On that note, do we want to just use the logging library for these messages instead of cerr prints?

efrederic commented on dummy-filesystem/dummy-filesystem.cc:
```diff
@@ -0,0 +1,96 @@
+
+	// Copy file into buffer
+	size_t bytesRead = fread(blk, 1, blockSize, file);
+	if (bytesRead == 0) {
```
If we hit this case and don't free blk then it looks like there is a memory leak

efrederic commented on dummy-filesystem/dummy-filesystem.cc:
```diff
@@ -0,0 +1,96 @@
+	std::string filename = "block" + oss.str() + ".txt";
+
+	blockMap[id] = filename;
+	std::ofstream myfile (filename);
```
"If the file cannot be opened, the stream's failbit flag is set." Should we be checking this for error opening the file?

pelmers commented:
`allocateBlock` and `getBlock` take or return char pointers. Since it's c++ I would prefer we pass around either std::string or std::vector so we avoid manual memory management and out of bounds.

Also `allocateBlock` and `rmBlock` return an int where they could be more clear with bool since they are pass/fail.

And the .h file specifies methods on a class but this class is never defined in the .cc, rather we have bare functions not attached to an object.
- Move blockMap to the `private:` part of FileSystem class. 
- Define the methods by changing `int allocateBlock(long id, unsigned char* blk) {}` to `int FileSystem::allocateBlock(long id, unsigned char* blk) {}` for each of the three methods
- The test will no longer #include filesystem.cc, only filesystem.h

Commit:
code review changes

Commit:
Merge branch 'group2-filesystem-mock' of https://github.com/Rice-Comp413-2016/RDFS into group2-filesystem-mock

Commit:
Finished native filesystem code review changes, ie:
wrote errors to log, closed file on error during reading, and now doesn't
erase block from map when there's an error deleting it
